src/mongoChangeStream.test.ts

```diff
@@ -1208,4 +1265,48 @@ describe.sequential('syncing', () => {
     sync.emitter.emit('foo', 'bar')
     assert.strictEqual(emitted, 'bar')
   })
+
+  test('should update the resume token, even if there were no records updated', async () => {
+    const { coll, db, redis } = await getConns()
+    const sync = await getSync()
+    await initState(sync, db, coll)
+    const processRecords = async () => {
+      await setTimeout(5)
+    }
+
+    // Utilities for comparing resume tokens to see if they change over time.
+    let lastToken: string | null = null
+    const { changeStreamTokenKey } = getKeys(coll, {})
+    const getCurrentToken = async () => await redis.get(changeStreamTokenKey)
+    const assertResumeTokenUpdated = async (when: string) =>
+      await assertEventually(
+        async () => (await getCurrentToken()) !== lastToken,
+        { message: `Resume token was not updated ${when}` }
+      )
+
+    // The resume token should initially be null, since we ran `initState`
+    // above.
+    lastToken = await getCurrentToken()
+    assert.equal(lastToken, null)
+
+    const changeStream = await sync.processChangeStream(processRecords)
+    changeStream.start()
+    // Let change stream connect
+    await setTimeout(ms('1s'))
```

Comment:
If there's a function we can call here that returns true if the change stream is connected, we could use that to replace this sleep with:

```typescript
await assertEventually(thatFunction, 'Change stream failed to connect')
```

Which would be better, for two reasons:

1. It would prevent a test case failure in the event that it takes more than 1 second to connect.
2. If it takes less than 1 second to connect, we could proceed right away instead of having to wait 1 second.